x/subscription/keeper/msg_server_auto_renewal.go

```diff
@@ -12,22 +13,73 @@ import (
 func (k msgServer) AutoRenewal(goCtx context.Context, msg *types.MsgAutoRenewal) (*types.MsgAutoRenewalResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	sub, found := k.GetSubscription(ctx, msg.Creator)
+	// Find consumer's subscription
+	sub, found := k.GetSubscription(ctx, msg.Consumer)
 	if !found {
 		return nil, utils.LavaFormatWarning("could not change auto-renewal of subscription", fmt.Errorf("subscription not found"),
-			utils.Attribute{Key: "sub_consumer", Value: msg.Creator},
+			utils.Attribute{Key: "consumer", Value: msg.Consumer},
+		)
+	}
+
+	// Verify creator either sub.Creator or sub.Consumer
+	if msg.Creator != sub.Consumer && msg.Creator != sub.Creator {
+		return nil, utils.LavaFormatWarning("could not change auto-renewal of subscription", fmt.Errorf("creator is not authorized to change auto-renewal for this subscription"),
+			utils.Attribute{Key: "creator", Value: msg.Creator},
+			utils.Attribute{Key: "consumer", Value: msg.Consumer},
 		)
 	}
 
-	sub.AutoRenewal = msg.Enable
-	err := k.subsFS.AppendEntry(ctx, msg.Creator, sub.Block, &sub)
+	// If msg.Enable == false, verify not already disabled
+	if !msg.Enable && sub.AutoRenewalNextPlan == types.AUTO_RENEWAL_PLAN_NONE {
+		return nil, utils.LavaFormatWarning("could not change auto-renewal of subscription", fmt.Errorf("auto-renewal is already disabled"),
+			utils.Attribute{Key: "creator", Value: msg.Creator},
+			utils.Attribute{Key: "consumer", Value: msg.Consumer},
+		)
+	}
+
+	// If msg.Enable == true, verify plan index
+	if msg.Enable {
+		if strings.TrimSpace(msg.Index) == "" {
```

Comment:
why do we need to trim spaces?

Reply:
Just in case? I wanted to see if the given arg is an empty string